### PR TITLE
Either async swap

### LINKED ISSUE #177: Proposition of a method `toEitherAsyncSwap` on maybeAsync
First, thank you first this great library !

When we want to check the existence of something before going on with the program, we can use `MaybeAsync` and then `.toEitherAsync`

I found myself with the opposite case. I want to check that something is not here before going on. Consider the following :

```typescript
type Person = { id: number };

function findPerson(id: number): MaybeAsync<Person>;
function save(person: Person): EitherAsync<Error, Person>;

const create = (person: Person): EitherAsync<Error, Person> => {
  const maybeAsyncPerson = findPerson(person.id);

  return maybeAsyncPerson
    .toEitherAsyncSwaped(new Error("Cannot create Person which already exists"))
    .chain(() => save(person));
};
```

I would be happy to make a PR with this new method `toEitherAsyncSwaped` if you find it interesting.

We can also imagine a `swap` on `eitherAsync`, which would be more generic. Then the previous case would become :

```typescript
const create = (person: Person): EitherAsync<Error, Person> => {
  const maybeAsyncPerson = findPerson(person.id);

  return maybeAsyncPerson
    .toEitherAsync(new Error("Cannot create Person which already exists"))
    .swap()
    .chain(() => save(person));
}
```

Maybe there is a different way that I did not think of...

### PR DESCRIPTION
**Pull request by JeromeBu**
closes #177

### CODE REVIEW

File: src/EitherAsync.ts [hunk 1]
```diff
@@ -111,6 +113,12 @@ class EitherAsyncImpl<L, R> implements EitherAsync<L, R> {
     })
   }
 
+  swap(): EitherAsync<R, L> {
+    return EitherAsync(async (helpers) => {
+      throw await this.runPromise((helpers as any) as EitherAsyncHelpers<L>)
```
Comment (JeromeBu) [inline src/EitherAsync.ts:118]: I am quite surprise by this implementation (inspired from mapLeft), but the tests passed with this...

Comment (gigobyte) [reply src/EitherAsync.ts:118]: This doesn't seem right, I'll have to check it out on my machine.

Comment (gigobyte) [reply src/EitherAsync.ts:118]: Turned out the tests for Either are not working at all. `Left` and `Right` have the same JSON value, so Jest can't disambiguate between them.

**Comment by gigobyte** [comment]
I've fixed the tests on master, if you rebase your test will fail.

**Comment by JeromeBu** [comment]
ok thanks

**Pull request merged**